tools/dynamo/gb_id_mapping.py

```diff
@@ -8,42 +8,60 @@
 
 
 def get_source_segment(source, node):
-    if hasattr(ast, "get_source_segment"):
-        return ast.get_source_segment(source, node)
+    return ast.get_source_segment(source, node)
 
 
-"""
-Normalizes string literals by removing formatting artifacts and escape sequences.
-Handles f-strings, quotes, newlines, and other syntax elements for cleaner output.
-"""
+def load_registry(path):
+    if path.exists():
+        with path.open() as f:
+            return json.load(f)
+    return {}
+
+
+def save_registry(reg, path):
+    with path.open("w") as f:
+        json.dump(reg, f, indent=2)
+
+
+def next_gb_id(reg):
+    ids = [int(x[2:]) for x in reg if x.startswith("GB") and x[2:].isdigit()]
+    return f"GB{(max(ids, default=0) + 1):04d}"
 
 
 def clean_string(s):
-    if s is None:
-        return None
+    """
+    Normalizes string literals by removing formatting artifacts and escape sequences.
+    Handles f-strings, quotes, newlines, and other syntax elements for cleaner output.
+    """
```

Comment:
Is it possible to reduce the number of regex substitutions here? e.g. there should be some builtin methods to remove escaped sequences.